### PR TITLE
Bump sinatra, thin, and poltergeist

### PR DESCRIPTION
**Pull request by girasquid**
Sinatra and Thin so that they can handle Rack 2.0, which Rails 5 requires. Poltergeist so that brew-installed phantomjs works.

@j-mutter @bslobodin

### CODE REVIEW

**Comment by girasquid** [comment]
The test suite...doesn't actually run for me using the updated poltergeist/phantomjs OR the old one; when you install using their README instructions the qless-core that you're testing against has problems like this all the way down:
```
  126) Qless::ClientEvents can pick up on pop events
     Failure/Error: client.jobs[queue.put(Qless::Job, { foo: 'bar' })]
     Redis::CommandError:
       ERR Error running script (call to f_3f9682e7ddb462dca8c60a26d5e88ac70c3a49e9): @user_script:925: @user_script: 925: Lua redis() command arguments must be strings or integers
```

File: Gemfile.lock [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
   remote: .
   specs:
-    qless (0.10.4)
+    qless (0.10.5)
       metriks (~> 0.9)
       redis (>= 2.2, < 4.0.0.rc1)
       rusage (~> 0.2.0)
       sentry-raven (~> 0.4)
-      sinatra (~> 1.3)
+      sinatra (= 2.0.0beta2)
```
Comment (bslobodin) [inline Gemfile.lock]: `beta` 😢

**Comment by girasquid** [comment]
[Known issue!](https://github.com/seomoz/qless-core/issues/63). Poking at downgrading my redis and experimenting.

File: Gemfile.lock (hunk 1 above)
Comment (girasquid) [inline Gemfile.lock]: From august https://rubygems.org/gems/sinatra/versions/2.0.0.beta2 😢

**Comment by girasquid** [comment]
This PR just bumps sinatra + thin instead of dealing with poltergeist at all: https://github.com/Shopify/qless/pull/6

**Pull request closed without merging**